src/app/Library/CrudPanel/CrudPanel.php

```diff
@@ -362,20 +370,26 @@ private function getRelationModelInstances($model, $relationString)
         $relationArray = explode('.', $relationString);
         $firstRelationName = array_first($relationArray);
         $relation = $model->{$firstRelationName};
+        $currentResults = [];
 
         $results = [];
-        if (! empty($relation)) {
-            if ($relation instanceof Collection) {
-                $currentResults = $relation->toArray();
+        if (! is_null($relation)) {
```

Comment:
It looks like the code quality took a hit:
![Screenshot 2020-02-24 at 08 46 39](https://user-images.githubusercontent.com/1032474/75133136-43e68580-56e2-11ea-8467-08dcd9f9ac0d.png)

Now, we don't _live by_ the scrutinizer grade, but it's usually right in indicating a method got a little too complicated.

Please try to include a little more comments inside these methods so it's easier to understand what's being done, and I'll see what I can do about improving code quality before merging.